test/widgets/profile_test.dart

```diff
@@ -0,0 +1,269 @@
+          0: ProfileFieldUserData(value: '[2]'),
+        }),
+        eg.user(userId: 2, fullName: 'test user'),
+      ];
+      final pushedRoutes = <Route<dynamic>>[];
+      final testNavObserver = TestNavigatorObserver()
+        ..onPushed = (route, prevRoute) => pushedRoutes.add(route);
+
+      await setupPage(tester,
+        users: users,
+        pageUserId: 1,
+        customProfileFields: [mkCustomProfileField(0, CustomProfileFieldType.user)],
+        navigatorObserver: testNavObserver,
+      );
+
+      final textFinder = find.text('test user');
+      check(textFinder.evaluate()).length.equals(1);
+      final fieldContainer = find.ancestor(of: textFinder, matching: find.byType(Column)).first;
+      final targetWidget = find.descendant(of: fieldContainer, matching:find.byType(Avatar));
+      await tester.tap(targetWidget, warnIfMissed: false);
+      check(pushedRoutes).last.isA<WidgetRoute>().page.isA<ProfilePage>().userId.equals(2);
+    });
+
+    testWidgets('page builds; dm links to correct narrow', (WidgetTester tester) async {
+      final pushedRoutes = <Route<dynamic>>[];
+      final testNavObserver = TestNavigatorObserver()
+        ..onPushed = (route, prevRoute) => pushedRoutes.add(route);
+      final expectedNarrow = DmNarrow(
+        allRecipientIds: [eg.selfUser.userId, 1]..sort(),
+        selfUserId: eg.selfUser.userId);
+
+      await setupPage(tester,
+        users: [eg.user(userId: 1)],
+        pageUserId: 1,
+        navigatorObserver: testNavObserver,
+      );
+
+      final targetWidget = find.byIcon(Icons.email);
+      await tester.tap(targetWidget, warnIfMissed: false);
+      check(pushedRoutes).last.isA<WidgetRoute>().page
+        .isA<MessageListPage>()
+        .narrow.equals(expectedNarrow);
+    });
+
+    testWidgets('page builds; user links render multiple avatars', (WidgetTester tester) async {
+      final users = [
+        eg.user(userId: 1, profileData: {
+          0: ProfileFieldUserData(value: '[2,3]'),
+        }),
+        eg.user(userId: 2, fullName: 'test user1'),
+        eg.user(userId: 3, fullName: 'test user2'),
+      ];
+
+      await setupPage(tester,
+        users: users,
+        pageUserId: 1,
+        customProfileFields: [mkCustomProfileField(0, CustomProfileFieldType.user)],
+      );
+
+      final findAvatars = find.byType(Avatar);
+      check(findAvatars.evaluate()).length.equals(2 + 1);
+    });
+
+    testWidgets('page builds; ensure long values do not overflow', (WidgetTester tester) async {
+      final longString = List.filled(400, 'X').join();
+      final user = eg.user(userId: 1, fullName: longString, profileData: {
+        0: ProfileFieldUserData(value: longString),
+        1: ProfileFieldUserData(value: '[1]'),
+      });
+
+      await setupPage(tester, users:[user], pageUserId: user.userId, customProfileFields: [
+        mkCustomProfileField(0, CustomProfileFieldType.shortText),
+        mkCustomProfileField(1, CustomProfileFieldType.user),
```

Comment:
Let's cover all the field types here (except perhaps `date` which can't be long with realistic data).

Particularly key is to cover links, since that's a separate widget; but really it's just an extra two or three lines to cover each type, so might as well be complete.